Performing C SOURCE FILE Test CMAKE_HAVE_LIBC_PTHREAD failed with the following output:
Change Dir: /mnt/d/taichi/taichi-1.5.0-patcher/taichi-1.5.0/_skbuild/linux-x86_64-3.8/cmake-build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make -f Makefile cmTC_40b4e/fast && /usr/bin/make  -f CMakeFiles/cmTC_40b4e.dir/build.make CMakeFiles/cmTC_40b4e.dir/build
make[1]: Entering directory '/mnt/d/taichi/taichi-1.5.0-patcher/taichi-1.5.0/_skbuild/linux-x86_64-3.8/cmake-build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_40b4e.dir/src.c.o
/usr/bin/cc -DCMAKE_HAVE_LIBC_PTHREAD  -fPIE -MD -MT CMakeFiles/cmTC_40b4e.dir/src.c.o -MF CMakeFiles/cmTC_40b4e.dir/src.c.o.d -o CMakeFiles/cmTC_40b4e.dir/src.c.o -c /mnt/d/taichi/taichi-1.5.0-patcher/taichi-1.5.0/_skbuild/linux-x86_64-3.8/cmake-build/CMakeFiles/CMakeTmp/src.c
Linking C executable cmTC_40b4e
/home/turbodamper/.local/lib/python3.8/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_40b4e.dir/link.txt --verbose=1
/usr/bin/cc CMakeFiles/cmTC_40b4e.dir/src.c.o -o cmTC_40b4e 
/usr/bin/ld: CMakeFiles/cmTC_40b4e.dir/src.c.o: in function `main':
src.c:(.text+0x22): undefined reference to `pthread_create'
/usr/bin/ld: src.c:(.text+0x2b): undefined reference to `pthread_detach'
/usr/bin/ld: src.c:(.text+0x34): undefined reference to `pthread_cancel'
/usr/bin/ld: src.c:(.text+0x41): undefined reference to `pthread_join'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [CMakeFiles/cmTC_40b4e.dir/build.make:100: cmTC_40b4e] Error 1
make[1]: Leaving directory '/mnt/d/taichi/taichi-1.5.0-patcher/taichi-1.5.0/_skbuild/linux-x86_64-3.8/cmake-build/CMakeFiles/CMakeTmp'
make: *** [Makefile:127: cmTC_40b4e/fast] Error 2


Source file was:
#include <pthread.h>

static void* test_func(void* data)
{
  return data;
}

int main(void)
{
  pthread_t thread;
  pthread_create(&thread, NULL, test_func, NULL);
  pthread_detach(thread);
  pthread_cancel(thread);
  pthread_join(thread, NULL);
  pthread_atfork(NULL, NULL, NULL);
  pthread_exit(NULL);

  return 0;
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /mnt/d/taichi/taichi-1.5.0-patcher/taichi-1.5.0/_skbuild/linux-x86_64-3.8/cmake-build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make -f Makefile cmTC_853da/fast && /usr/bin/make  -f CMakeFiles/cmTC_853da.dir/build.make CMakeFiles/cmTC_853da.dir/build
make[1]: Entering directory '/mnt/d/taichi/taichi-1.5.0-patcher/taichi-1.5.0/_skbuild/linux-x86_64-3.8/cmake-build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_853da.dir/CheckFunctionExists.c.o
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create -fPIE -MD -MT CMakeFiles/cmTC_853da.dir/CheckFunctionExists.c.o -MF CMakeFiles/cmTC_853da.dir/CheckFunctionExists.c.o.d -o CMakeFiles/cmTC_853da.dir/CheckFunctionExists.c.o -c /home/turbodamper/.local/lib/python3.8/site-packages/cmake/data/share/cmake-3.21/Modules/CheckFunctionExists.c
Linking C executable cmTC_853da
/home/turbodamper/.local/lib/python3.8/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_853da.dir/link.txt --verbose=1
/usr/bin/cc  -DCHECK_FUNCTION_EXISTS=pthread_create CMakeFiles/cmTC_853da.dir/CheckFunctionExists.c.o -o cmTC_853da  -lpthreads 
/usr/bin/ld: cannot find -lpthreads
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [CMakeFiles/cmTC_853da.dir/build.make:100: cmTC_853da] Error 1
make[1]: Leaving directory '/mnt/d/taichi/taichi-1.5.0-patcher/taichi-1.5.0/_skbuild/linux-x86_64-3.8/cmake-build/CMakeFiles/CMakeTmp'
make: *** [Makefile:127: cmTC_853da/fast] Error 2



